### PR TITLE
Add an option to totally disable doctrine support

### PR DESCRIPTION
**Pull request by odolbeau**
For now it's not possible to use Api Platform without doctrine easily as the doctrine service is used to declare several services.

### CODE REVIEW

**Comment by odolbeau** [comment]
I think it could be idea to use tagged services for type extractors & loaders.
Let me know if you want me to do it. :)

**Comment by sroze** [comment]
Hey @odolbeau! For 1.0 that's a good point, but it isn't for 2.0 as we will use the new Symfony PropertyInfo component which "have" its Doctrine extractor only in the Doctrine Bridge.

By the way, for 1.0 that would be useful. Instead of adding a configuration, can you instead guess it from the non-existence of a Doctrine service? Then, could you base your PR on 1.0 branch?

**Comment by odolbeau** [comment]
I probably need to give a try to the 2.0 version or at least, take a look a the code. :)

I'm not a big fan of checking the existence of a service as it can be a valid use case to use doctrine in an application but not with the API Platform. Anyway, in my case, I don't use doctrine at all so yes, I'll update my code. :)

**Comment by sroze** [comment]
That makes sense. In that case, setting the default value of this parameter
with the existence of this Doctrine service maybe?

**Comment by odolbeau** [comment]
I have absolutely no idea of how changing a parameter default value regarding the existence of a service. :/ Is it possible?
Regarding the first option, it's obviously not possible to check doctrine service existence inside the `load` method of the extension. I can check if the `doctrine` extension exist in the `prepend` method and store it in a property. It means the Extension will have a context.
Do you see a better way of doing this?

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -34,6 +34,7 @@ public function getConfigTreeBuilder()
                 ->scalarNode('title')->cannotBeEmpty()->isRequired()->info('The title of the API.')->end()
                 ->scalarNode('description')->cannotBeEmpty()->isRequired()->info('The description of the API.')->end()
                 ->scalarNode('cache')->defaultFalse()->info('The caching service to use. Set to "dunglas_api.mapping.cache.apc" to enable APC metadata caching.')->end()
+                ->booleanNode('enable_doctrine_orm')->defaultValue(true)->info('Enable the Doctrine ORM integration.')->end()
```
Comment (dunglas) [inline DependencyInjection/Configuration.php]: Here you can test if the a class from the Doctrine bridge exist to set the default value.

File: composer.json [hunk 2]
```diff
@@ -19,9 +19,6 @@
     "symfony/framework-bundle": "~2.6|~3.0",
     "symfony/proxy-manager-bridge": "~2.3",
     "ocramius/proxy-manager": "~1.0",
-    "doctrine/orm": "~2.2,>=2.2.3",
-    "doctrine/inflector": "~1.0",
```
Comment (dunglas) [inline composer.json]: The inflector is always required even if you don't use the ORM.

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (odolbeau) [inline DependencyInjection/Configuration.php]: Genious! <3

**Comment by dunglas** [comment]
:+1: ping @api-platform/core-team

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -29,11 +29,15 @@ public function getConfigTreeBuilder()
         $treeBuilder = new TreeBuilder();
         $rootNode = $treeBuilder->root('dunglas_api');
 
+        // We chack if symfony's doctrine bridge is enabled or not to choose the defaultValue for `enable_doctrine_orm`.
```
Comment (sroze) [inline DependencyInjection/Configuration.php]: chack like shake your body? :smile:

**Comment by sroze** [comment]
Looks great @odolbeau, thank you!

**Pull request merged**